src/languages/csharp.js

```diff
@@ -177,13 +178,16 @@ export default function(hljs) {
       },
       {
         className: 'function',
-        begin: '(' + TYPE_IDENT_RE + '\\s+)+' + hljs.IDENT_RE + '\\s*\\(', returnBegin: true,
+        begin: '(' + TYPE_IDENT_RE + '\\s+)+' + hljs.IDENT_RE + '\\s*(\\<.+\\>\\s*)?\\(', returnBegin: true,
```

Comment:
I think the trailing `\\s*` should be outside the optional match, yes?

Reply:
There is no difference in result, but it is better for grouping. Ok, I'll modify it.